scripts/affiliate_server.py

```diff
@@ -343,6 +369,16 @@ def to_dict(self):
 class Submit:
     @classmethod
     def unpack_isbn(cls, isbn) -> tuple[str, str]:
+        """
+        Return an "ASIN" and an ISBN 13 if possible. Here, an "ASIN" is an
+        ISBN 10 if the input is an ISBN, and an Amazon-specific ASIN starting
+        with "B" if it's not an ISBN 10.
+        """
+        # If the "isbn" is an Amazon-specific ASIN, return it.
+        if len(isbn) == 10 and isbn.startswith("B"):
+            return (isbn, "")
+
+        # Get ISBN 10 and 13.
         isbn = normalize_isbn(isbn.replace('-', ''))
         isbn10 = (
```

Comment:
Blocker?

ISBN_10 can always be converted to ISBN_13, however, there's a class of ISBN_13 that cannot (by definition of the need to switch to isbn13) be converted back to ISBN_10.

In this case, what happens in the cases where we return (None, ISBN_13)? Does this break anything?

Reply:
This is taken care of.